core/db/migrate/20220113052823_create_payment_sources.rb

```diff
@@ -0,0 +1,21 @@
+class CreatePaymentSources < ActiveRecord::Migration[5.2]
+  def change
+    create_table :spree_payment_sources do |t|
+      t.string :gateway_payment_profile_id, index: { unique: true }
```

Comment:
@GrzegorzStochel this unique index should be in conjunction with type, there is a slight chance that the ID could be the same for 2 different gateways

Reply:
done